linked_list/main.py

```diff
@@ -43,53 +43,173 @@ class Node(object):
     Node class representing each of the linked nodes in the list.
     """
 
-    def __init__(self, elem, next=None):
-        pass
+    def __init__(self, elem, next=None): # jon:DONE
+        self.elem = elem
+        self.next = None
 
-    def __str__(self):
-        pass
-
-    def __eq__(self, other):
-        pass
+    def __str__(self): # jon:DONE
+        return str(self.elem)
 
-    def __repr__(self):
-        pass
+    def __eq__(self, other): # akshith: Done
+        if type(self) == type(other):
```

Comment:
In Python we usually try to avoid checking types. We have a more duck-typing focus. Please remind me to explain this during next class.